structuralcodes/materials/concrete/_concreteEC2_2023.py

```diff
@@ -0,0 +1,195 @@
+        )
+        self._kE = kE
+        self._concrete_dev_class = concrete_dev_class
+        self._gamma_C = gamma_C
+
+        # TODO: implement this method
+        def _reset_attributes(self):
+            # We only need to reset computed properties
+            self._fcm = None
+            self._fctm = None
+            self._Ecm = None
+
+        @property
+        def fcm(self) -> float:
+            """Returns the mean strength of concrete.
+
+            Returns:
+                float: The mean compressive strength in MPa.
+            """
+            if self._fcm is not None:
+                return self._fcm
+            self._fcm = ec2_2023.fcm(self.fck)
+            return self._fcm
+
+        @fcm.setter
+        def fcm(self, value: float):
+            """Sets a user defined value for the mean strength of concrete.
+
+            Args:
+                value (float): the value of the mean strength of
+                    concrete in MPa.
+
+            Raises:
+                ValueError: if value is less or equal than the value of fck
+            """
+            if abs(value) <= self._fck:
+                raise ValueError(
+                    (
+                        'Mean compressive strength cannot be lower than',
+                        'characteristic strength.\n',
+                        'Current characteristing strength: ',
+                        f'fck = {self._fck}.',
+                        f'Current value: value = {value}',
+                    )
+                )
+            self._reset_attributes()
+            self._fcm = value
+
+        @property
+        def fctm(self) -> None:
+            """Returns the mean concrete tensile strength.
+
+            Returns:
+                float: the mean concrete tensile strength.
+            """
+            if self._fctm is not None:
+                return self._fctm
+            self._fctm = ec2_2023.fctm(self.fck)
+            return self._fctm
+
+        @fctm.setter
+        def fctm(self, value: float):
+            """Sets a custom user defined value for the concrete tensile
+                strength for the concrete.
+
+            Args:
+                value (float): the new value for fctm in MPa.
+            """
+            self._reset_attributes()
+            self._fctm = value
+
+        @property
+        def fctk_5(self) -> float:
+            return ec2_2023.fctk_5(self.fctm)
+
+        @property
+        def fctk_95(self) -> float:
+            return ec2_2023.fctk_95(self.fctm)
+
+        @property
+        def Ecm(self) -> float:
+            if self._Ecm is not None:
+                return self._Ecm
+            self._Ecm = ec2_2023.Ecm(self.fcm, self._kE)
+            return self._Ecm
+
+        @Ecm.setter
+        def Ecm(self, value: float) -> None:
+            self._reset_attributes()
+            self._Ecm = value
+
+        def fcd(self, t_ref: float, t0: float, fck_ref: float = 40) -> float:
```

Comment:
I have a doubt. With this function we can compute design strength of concrete developed during time.
Most of the times I think users just need fcd (28 days). Could't we make the time-related parameter as optional with default values in order to have by default from fcd the design strength at 28 days?

Reply:
Done! I have also added the same optional argument for `fctd`

Reply:
I like that you have changed the signature such that the function represents the equation it implements. This way the user can choose to calculate the arguments with the other implemented functions, or simply provide other values directly 👍